### ISSUE TITLE
Iteration Plan for February 2018

### ISSUE THREAD

**Issue by kieferrm** [Microsoft/vscode#43361]
This plan captures our work in **February**. After three 5-week iterations we are back to having a 4 week iteration. We will ship **early March** and call it **February Update** following our (in)famous naming scheme. The focus will be on finishing up work we started in January and kicking off a few new work areas. Among them will be the initial investigation into terminal splitting, supporting more flexible workbench layouts, and language aware folding. All three items are among the top upvoted features (4, 5, 10).

## Endgame
- **Feburary 26th**: Code freeze for the endgame
- **March 2nd**: Endgame done

The endgame details for this iteration are tracked in #44419

## Plan Items
Below is a summary of the top level plan items.

Legend of annotations:

| Mark | Description |
| ------------- | ------------- |
| :runner: | work in progress |
| :hand: | blocked task |
| :muscle: | stretch goal for this iteration |
| :red_circle: | missing issue reference |
| :large_blue_circle: | more investigation required to remove uncertainty |
| :black_circle: | under discussion within the team |

### Install
- [ ] :runner: Improve the update experience on Windows #41676 @joaomoreno
- :hand: :runner: Allow user level installs on Windows @joaomoreno
- :hand: :runner: Provide a VS Code snap package #5458 @tyriar
- :hand: Support 32-bit apt repositories #20790 @tyriar

### Workbench
- [x] Improve notification UI #22388 @bpasero
- [x] Investigate showing search results in a panel #9435 @isidorn
- [x] Recommend language packs based on an user's locale #44012 @ramya-rao-a
- [x] Language packs - cache cleanup @dbaeumer
- [x] Enable users of the insider build to easly provide feedback on setting search #43268 @roblourens
- [x] Indicate symbolic links in the explorer  #11539 @isidorn
- [x] Centered Zen mode presentation for single editors #15684 @isidorn

### Editor
- [x] Finalize text model and storage reimplementation to improve performance #41042 @alexandrudima @rebornix
- [ ] :runner: Ship a new version of the monaco editor [See query.](https://github.com/Microsoft/monaco-editor/issues?q=is%3Aopen+is%3Aissue+milestone%3A%22On+Deck%22) @rebornix

### Terminal
- [x] Investigate Terminal splitting (`tmux` without `tmux`) #7504 @tyriar

#### Language Server Protocol
- [x] Support additional request kinds in completion items @dbaeumer @ramya-rao-a
- [x] Promote proposed protocol features (Configuration, Workspace Folder, Color Provider) @dbaeumer @aeschli

#### Emmet
- [x] Improve emmet activation in `html` and `css` files #29113 @ramya-rao-a
- [x] Provide css emmet abbr suggestions when inside style attribute #29066 @gushuro
- [ ] :runner: Wrap selection with abbreviation in real time #23169 @gushuro @ramya-rao-a

#### JavaScript/TypeScript
- [ ] :runner: Language aware folding (preview) @mjbvz

#### CSS/HTML
- [x] Add intelisense support for `src ` attributes and `href` in html #2037 @octref @aeschli
- [x] HTML - Language aware folding (preview) @aeschli

### Extensions
- [x] Support to disable built-in extensions #17068 @sandy081
- [x] Support that an extension can do additional steps when it is uninstalled #35006 @sandy081
- [x] Add `Manage Extensions` to the activity bar gear #43708 @ramya-rao-a
- [x] Setting to only show extension recommendations on demand #43471  @ramya-rao-a

### Extension Contributions
- [ ] :runner: Vetur: Review PR for JS completions in template section of a `vue` file vuejs/vetur#681 @octref

### Debug
- [x] Explore how to support LogPoints #42965 @weinand @isidorn
- [x] Show breakpoint type and status in breakpoints viewlet #42266 #3388 #7262 @isidorn @weinand
- [x] Explore auto attach debugger to Node processes started from integrated terminal #42521 @weinand
- Debt
   - [x] Support to cleanup up Docker/nodemon/WSL processes #38424 #28108 #19203 @weinand @roblourens
   - :hand: How to handle `Promise.reject` discussion #1746 @weinand @roblourens @isidorn

### API
- [x] Finalize Breakpoints API #43492 @weinand
- [x] Proposed API: Introduce support for language-aware folding @aeschli
- [ ] :runner: Proposed API for a revamped HTMLPreview #43713 @mjbvz
- [x] Provide API that hints at the top most visible line in the editor #44424 @alexandrudima
- [x] Proposed API for TreeView for selection/reveal #30288 #28338 @sandy081

### Performance
- [x] Use ASAR for bundled node modules (without breaking extensions) #36997 @alexandrudima
- [x] Explore use of V8 snapshots to improve startup performance @alexandrudima

### Serviceability
- [x] Polish issue reporter UI and experience #43518 @rmacfarlane
- [x] Avoid hitting GH rate limits when searching duplicates #43519 @rmacfarlane @chrmarti
- [ ] :runner: :muscle: Process explorer as a separate renderer window #41045 @rmacfarlane

### Engineering
- [x] Improve documentation of our electron upgrade process #41036 @tyriar
- [ ] :runner: Early adoption testing for electron 2.0 beta @tyriar @bpasero
- [x] Auto-enable/disable GH bots based on the phase of the iteration #43725 @chrmarti
- [x] Self-host on `@ts-check` for our JS code #41678 #43973 @egamma @joaomoreno @dbaeumer
- [x] Re-enable unused variable check in hygiene checks #42831 @egamma @joaomoreno
- [x] Experimental support for searching duplicate GH issues #43728 @chrmarti

### Documentation
- [x] Docathon on 2/16 and 2/19 Microsoft/vscode-docs#1396 **team**

-------
### Deferred Items
2/26
- Investigate support a grid layout for editors #14909 @stevencl @bpasero @tyriar @joaomoreno
- Improve settings experience #41040 @sandy081 @roblourens
- Investigate promoting explorer contributions to viewlets #43645 @sandy081
- LSP: Propose support for language aware folding #3422  @aeschli
- LSP: Support code action types Microsoft/language-server-protocol#389 @dbaeumer
- Show visual hints for column breakpoints #31612 @isidorn @weinand
- Adopt logging service #41680 **team** @joaomoreno
- Propose support for multi-location diagnostics @kieferrm @dbaeumer @jrieken
- Investigate npm module for LS single file tests Microsoft/vscode-languageserver-node#312 @octref
- Improve auto-activation of intellisense in `jsdoc` types #43715 @mjbvz
- Improve auto-activation of intellisense in template literals #37176 @mjbvz

2/19
- Optimize 'find' for large files @rebornix
- Improve speed of the expand command by replacing emmet parser #43470 @gushuro @ramya-rao-a
- Provide setting for visually indicating whitespace in selections #1477 @ramya-rao-a
- Finalize logging API to extensions #43276 @roblourens
- Adopt TS 2.7 for selfhosting #42853 @mjbvz - blocked by tslint not being compatible with ts 2.7
- Provide project testing support for `css` and `html` language servers @octref

**Comment by NateLing** [comment]
Hi,guys. any plan to support webassembly extension?

**Comment by RoyTinker** [comment]
Suggest #10121 as primary issue reference for "Explore UX for flexible workbench layout" (note it's now the highest by upvotes and leads the other `workbench-layout` issues by over 600 votes).

**Comment by SailorMax** [comment]
@ramya-rao-a "Recommend language packs based on an user's locale"
Not user's locale, but user's OS UI language!
If I live in Russia, but my OS language is English => I prefer English UI and I don't need recommendations about Russian UI. Locale != UI language.

**Comment by patrys** [comment]
"Locale" is often short for "language, number format, date format and so on".

**Comment by SailorMax** [comment]
@patrys In theory you right.
On practice I have many samples in MS apps with broken logic on this theme...
My comment was "warning" :)

**Comment by mifas** [comment]
Please try to release this soon #39307 #40255

**Comment by Tyriar** [comment]
@SailorMax [`app.getLocale`](https://electronjs.org/docs/api/locales) is probably the call that will be used, which calls into [`l10n_util::getApplicationLocale`](https://cs.chromium.org/chromium/src/ui/base/l10n/l10n_util.h?type=cs&sq=package:chromium&l=45)

**Comment by SailorMax** [comment]
@Tyriar In result will be country, but not OS UI language. Many countries has few official languages. Which one will be recommend based on country? And why, if I don't use any from them on my PC?
Based on app.getLocale we will get +1 program with broken logic :)

**Comment by aberonni** [comment]
As @RoyTinker could we link the UX discovery task to an issue? I also suggest #10121 or maybe #14909 , otherwise a new issue that we could link into those issues to explain to people that something is happening.

**Comment by kczx3** [comment]
Are there any plans to work on moving the [Remote File System API](https://code.visualstudio.com/updates/v1_17#_preview-remote-file-system-api) forward?

**Comment by DarkLite1** [comment]
Suggest #3422 as primary issue reference for " Introduce support for language-aware folding @aeschli" and "Propose support for language aware folding @aeschli"

**Comment by joshj** [comment]
Thank you for looking into #14909! I am really excited to see some movement!

**Comment by zeferinix** [comment]
#14909 Finally!
